lib/ansible/galaxy/api.py

```diff
@@ -817,8 +817,20 @@ def get_collection_version_metadata(self, namespace, name, version):
 
         signatures = data.get('signatures') or []
 
+        download_url_info = urlparse(data['download_url'])
+        if not download_url_info.scheme and not download_url_info.path.startswith('/'):
+            # galaxy does a lot of redirects, with much more complex pathing than we use
+            # within this codebase, without updating _call_galaxy to be able to return
+            # the final URL, we can't reliably build a relative URL.
```

Comment:
I was wondering if this block could be simplified as just `download_url = urljoin(self.api_server, data['download_url'])`, but I guess maybe not due to this.

Reply:
Not supporting relative paths seems reasonable. I was wondering if this error was excessive, but it's better than constructing the wrong URL and letting the user decipher the cause. I'll clarify my other suggestion on the relevant lines.